### DISCUSSION TITLE
Trying to create an accessory that would display a RED/GREEN light based on pin 23 status.

### DISCUSSION THREAD

**Discussion by AndreCouture** [HomeSpan/HomeSpan#706] [Q&A]
Hello,

I've been trying to get a very simple status returned to my HomeKit but having some issues.

The code is simply, it read the status of pin 23 and should display a RED light in my Home Accessory when HIGH and GREEN when LOW.

Basically I will use this to tell me when my Hydro tariff changes.

Here's my code;

`#include <HomeSpan.h>

class StatusLight : public Service::LightBulb {
private:
    int statusPin = 23; // GPIO pin to monitor
    int led = 2; // ESP32 Pin to which onboard LED is connected
    Characteristic::Hue *hue;
    Characteristic::Saturation *saturation;

public:
    StatusLight() : Service::LightBulb() {
        new Characteristic::On();
        hue = new Characteristic::Hue();
        saturation = new Characteristic::Saturation();
        pinMode(statusPin, INPUT); // Initialize the status pin as an input
    }

    void loop() {
        bool pinStatus = digitalRead(statusPin);

        if (pinStatus) { // Pin is HIGH, set light to red
            hue->setVal(0); // Hue value for red
            saturation->setVal(100); // Full saturation for solid color
            digitalWrite(led,  HIGH);
        } else { // Pin is LOW, set light to green
            hue->setVal(120); // Hue value for green
            saturation->setVal(100); // Full saturation for solid color
            digitalWrite(led,  LOW);
        }
    }
};

void setup() {
  Serial.begin(115200);
  homeSpan.begin();

  new SpanAccessory();
  new StatusLight();
}

void loop() {
  homeSpan.poll();
}
`

When adding the accessory I get a message
"Unable to Add Accessory" Accessory is out of compliance.

What is it and how to fix it

Thanks

**Comment by HomeSpan** [comment]
Your set-up is missing key information (like the Identify Characteristic) and your LightBulb is missing an ability to turn the light on/off.   Some of these error messages should have appeared directly on the Serial Monitor when you tried to run the sketch. Please review the tutorial examples for details on how to create HomeSpan sketches.

**Comment by HomeSpan** [comment]
Also, if you are trying to display a red/green light on the Home App itself, as a form of an alert, this probably won't yield the desired result.  The Home App is designed to allow you to set colors.  You can certainly see the color that is set but you have to navigate through a number of tiles and pages fore this to appear.  If you are looking to be a device that can send an alert to the Home App, you should consider the contact switch, or even the programmable stateless service.

**Comment by AndreCouture** [comment]
Thanks I got a bit further now by adding the characteristics Why is my accessory from the Home Screen always show as no response?I’m using based on a an example provided with the package 1.8.0—Regards,Andre CoutureSenior Cloud Solution ***@***.***://www.linkedin.com/in/andrecoutureOn Dec 2, 2023, at 18:38, HomeSpan ***@***.***> wrote:﻿
Your set-up is missing key information (like the Identify Characteristic) and your LightBulb is missing the required Characteristics.   These error messages should have appeared directly on the Serial Monitor when you tried to run the sketch. Please review the tutorial examples for details on how to create HomeSpan sketches.

**Comment by AndreCouture** [comment]
My bad!  My AppleTV was on a different network hence the not responding issue!Thank you for your help!—Regards,Andre CoutureSenior Cloud Solution ***@***.***://www.linkedin.com/in/andrecoutureOn Dec 2, 2023, at 19:45, Andre Couture ***@***.***> wrote:﻿Thanks I got a bit further now by adding the characteristics Why is my accessory from the Home Screen always show as no response?I’m using based on a an example provided with the package 1.8.0—Regards,Andre CoutureSenior Cloud Solution ***@***.***://www.linkedin.com/in/andrecoutureOn Dec 2, 2023, at 18:38, HomeSpan ***@***.***> wrote:﻿
Your set-up is missing key information (like the Identify Characteristic) and your LightBulb is missing the required Characteristics.   These error messages should have appeared directly on the Serial Monitor when you tried to run the sketch. Please review the tutorial examples for details on how to create HomeSpan sketches.
